### PR TITLE
fix: event fix and added utility

### PR DESCRIPTION
**Pull request by aashishmalik**
## What are the changes introduced in this PR?

Added a util function to ease the process of validating parameters and conversion to string.

Added util to check for 
1. event mandatory ?
2. event exists ?
3. event exists & not object
4. lowercase conversion

## What is the related Linear task?

Resolves INT-XXX

## Please explain the objectives of your changes below

Put down any required details on the broader aspect of your changes. If there are any dependent changes, **mandatorily** mention them here

### Any changes to existing capabilities/behaviour, mention the reason & what are the changes ?

N/A

### Any new dependencies introduced with this change?

N/A

### Any new generic utility introduced or modified. Please explain the changes.

N/A

### Any technical or performance related pointers to consider with the change?

N/A

@coderabbitai review

<hr>

### Developer checklist

- [ ] My code follows the style guidelines of this project

- [ ] **No breaking changes are being introduced.**

- [ ] All related docs linked with the PR?

- [ ] All changes manually tested?

- [ ] Any documentation changes needed with this change?

- [ ] Is the PR limited to 10 file changes?

- [ ] Is the PR limited to one linear task?

- [ ] Are relevant unit and component test-cases added?

### Reviewer checklist

- [ ] Is the type of change in the PR title appropriate as per the changes?

- [ ] Verified that there are no credentials or confidential data exposed with the changes.

### CODE REVIEW

**Comment by devops-github-rudderstack** [comment]
Test report for this run is available at: https://test-integrations-dev.s3.amazonaws.com/integrations-test-reports/rudder-transformer/3142/test-report.html

File: src/v0/util/index.js [hunk 1]
```diff
@@ -2201,6 +2201,26 @@ const combineBatchRequestsWithSameJobIds = (inputBatches) => {
   return combineBatches(combineBatches(inputBatches));
 };
 
+/**
+ * This function validates the event and return it as string.
+ * @param {*} isMandatory The event is a required field.
+ * @param {*} converToLowercase The event should be converted to lower-case.
+ * @returns {string} Event name converted to string.
+ */
+const validateEventAndLowerCaseConversion = (event, isMandatory, converToLowercase) => {
```
Comment (koladilip) [inline src/v0/util/index.js:2210]: converToLowercase convertToLowercase

File: src/v0/util/index.js [hunk 2]
```diff
@@ -2201,6 +2201,26 @@ const combineBatchRequestsWithSameJobIds = (inputBatches) => {
...
+  if (typeof event === 'object' || !isDefined(event)) {
+    throw new InstrumentationError('Event should not be a object, function or NaN and undefined');
+  }
+  if (!isMandatory) {
```
Comment (koladilip) [inline src/v0/util/index.js:2214]: if (isMandatory && event === 0 ) {
  throw new InstrumentationError('Event is a required field');
}

return converToLowercase ? event.toString().toLowerCase() : event.toString();

// restructure like this helps to reduce conditions

Comment (aashishmalik) [reply src/v0/util/index.js]: Not thorwing error if 0, as some destinations take integer strings and "0" will be valid

File: src/v0/util/index.js (hunk 1 above)
Comment (aashishmalik) [reply src/v0/util/index.js]: done

**Review by koladilip — DISMISSED** [review]

File: src/v0/util/index.js [hunk 3]
```diff
@@ -2201,6 +2201,25 @@ const combineBatchRequestsWithSameJobIds = (inputBatches) => {
...
 };
 
+/**
+ * This function validates the event and return it as string.
+ * @param {*} isMandatory The event is a required field.
+ * @param {*} convertToLowerCase The event should be converted to lower-case.
+ * @returns {string} Event name converted to string.
+ */
+const validateEventAndLowerCaseConversion = (event, isMandatory, convertToLowerCase) => {
+  if (typeof event === 'object' || !isDefined(event)) {
```
Comment (koladilip) [inline src/v0/util/index.js:2211]: Interchange these conditions

File: src/v0/util/index.test.js [hunk 4]
```diff
@@ -456,3 +456,44 @@ describe('Unit test cases for combineBatchRequestsWithSameJobIds', () => {
     expect(combineBatchRequestsWithSameJobIds(input)).toEqual(expectedOutput);
   });
 });
+
+describe('validateEventAndLowerCaseConversion Tests', () => {
```
Comment (koladilip) [inline src/v0/util/index.test.js:460]: add test case for boolean values, what should happen in that cases for true and false

Comment (aashishmalik) [reply src/v0/util/index.test.js:460]: it will throw error as our utility always converts to string

**Comment by yashasvibajpai** [comment]
@aashishmalik can we use [this](https://github.com/rudderlabs/rudder-transformer/blob/3e8790ae6e2cc2c33e9a2da21ec8c10a68b4219f/src/v0/util/index.js#L2084) general utility here?

**Comment by aashishmalik** [comment]
> @aashishmalik can we use [this](https://github.com/rudderlabs/rudder-transformer/blob/3e8790ae6e2cc2c33e9a2da21ec8c10a68b4219f/src/v0/util/index.js#L2084) general utility here?

we will remove this as this is basic utility

**Review by yashasvibajpai — APPROVED** [review]

**Review by koladilip — APPROVED** [review]

**Pull request merged**